### DISCUSSION TITLE
How do I change the pixel values of an image

### DISCUSSION THREAD

**Discussion by Bettygeuse** [robotology/yarp#3117] [Support - Q&A] [answered]
I'm having trouble editing the pixel values of an image, what am I doing wrong? From my testing, it seems that I can't actually set the values for the pixels because the changes aren't taking affect. I've tried printing out the rgb values of the pixels in the image and they return blank, not a zero integer, just blank. The yarp viewer returns a gray image as shown in the bottom picture (in the picture, there are 0 frames coming in from the port because it is disconnected, yarp viewer returns the same image even when connected to the input port with frames coming in). The code is below as well. 

![temp](https://github.com/robotology/yarp/assets/57381704/01eb933e-b87b-4153-a109-1268fc2a7c95)
![view](https://github.com/robotology/yarp/assets/57381704/f0a8dd6a-af53-4413-b948-7d846e314abe)

**Comment by pattacini** [comment]
Hi @Bettygeuse 

From what you posted, it seems you're manipulating the pixel correctly.
However, you missed posting the code that is relevant to sending the image out via a YARP port.

I can point you at some code snippet that is doing pretty much what you're looking for:
- https://github.com/vvv-school/tutorial_gaze-interface/blob/e5734462df4beae9103fc46870e7773f7438e2da/src/detector.cpp#L28

**Reply by Bettygeuse** [reply]
Hi @pattacini !

This is very useful, thank you! I'm currently taking a look at the file and am testing out some changes. From what I can see, it is a little different from what I am trying to do. Instead of reading in a yarp image from a different port and manipulating it, I am receiving an array containing the rgb values of pixels of an image. Using these values I'm hoping to recreate the image, and send it over a port. 

Alternatively, I was thinking of testing out the setExternal() method instead of manually assigning pixel values, but I would still be creating a yarp image instead of reading one in. My current hunch is that I made a mistake with setting up the ports. 

Thank you for your help

**Reply by Bettygeuse** [reply, marked as answer]
The setExternal() method was the solution! I had the bgra bytes of my image and I just needed to pass them into the method. Thank you for the help!
